From db643c4f0a04100125e27eb60a35808bfd124ec0 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 10 Oct 2022 12:21:19 +0300
Subject: [PATCH] [SW-0] gaudi3: add progress bits for boot info

Currently, progress bits for boot info set and get are added only
for error scenario, this patch adds start and finish indication
for boot info set and get for debug purpose.

Change-Id: Iac40fbff92330efe0a837fefd050cc9c601efbb2
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 6281d92..5cedf2f 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1235,6 +1235,8 @@
 		return;
 
 	if (!arcboot_rom_global.boot_info_set_done) {
+		arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START);
+
 		if (boot_info_set_no_merge(HL_BOOT_INFO_IMG_DESC_INFO, &p_img_ptr_ctx->desc_info,
 		   sizeof(struct hl_fw_desc_info))) {
 			arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FAIL);
@@ -1259,7 +1261,11 @@
 		cold_rst_data.data = regmap_readl(mmCOLD_RST_FLOPS_2);
 		cold_rst_data.boot_info_set_done = 1;
 		regmap_writel(cold_rst_data.data, mmCOLD_RST_FLOPS_2);
+
+		arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FINISH);
 	} else {
+		arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START);
+
 		if (boot_info_get_no_merge(HL_BOOT_INFO_IMG_DESC_INFO,
 			&p_img_ptr_ctx->desc_info,
 			sizeof(struct hl_fw_desc_info), &copy_sz)) {
@@ -1305,6 +1311,8 @@
 				}
 			}
 		}
+
+		arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH);
 	}
 }
 
diff --git a/arch/gaudi3/arc/progress_bits.h b/arch/gaudi3/arc/progress_bits.h
index 2d16bff..23bb399 100644
--- a/arch/gaudi3/arc/progress_bits.h
+++ b/arch/gaudi3/arc/progress_bits.h
@@ -42,128 +42,132 @@
 	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 35,
 	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 36,
 	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 37,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FAIL = 38,
-	PROGRESS_BOOT_INFO_IMG_DESC_DATA_SET_FAIL = 39,
-	PROGRESS_BOOT_INFO_SECURITY_DATA_SET_FAIL = 40,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FAIL = 41,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_INVALID = 42,
-	PROGRESS_BOOT_INFO_IMG_DESC_DATA_GET_FAIL = 43,
-	PROGRESS_BRINGUP_PCI_PRI = 44,
-	PROGRESS_BRINGUP_PCI_PRI_LOADADDR_INVALID = 45,
-	PROGRESS_BRINGUP_PCI_PRI_STARTADDR_INVALID = 46,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 47,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 48,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 49,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 50,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 51,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 52,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 53,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 54,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 55,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 56,
-	PROGRESS_BRINGUP_PCI_SEC = 57,
-	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 58,
-	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 59,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 60,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 61,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 62,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 63,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 64,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 65,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 66,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 67,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 68,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 69,
-	PROGRESS_ARCBOOT_IN_RECOVERY = 70,
-	PROGRESS_ARCBOOT_ROM_TPM = 71,
-	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 72,
-	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 73,
-	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 74,
-	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 75,
-	PROGRESS_BRINGUP_TPM_SPI = 76,
-	PROGRESS_BRINGUP_TPM_SPI_PRI = 77,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_LOADADDR_INVALID = 78,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_STARTADDR_INVALID = 79,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 80,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 81,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 82,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 83,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 84,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 85,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 86,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 87,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 88,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 89,
-	PROGRESS_BRINGUP_TPM_SPI_SEC = 90,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_LOADADDR_INVALID = 91,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_STARTADDR_INVALID = 92,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 93,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 94,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 95,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 96,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 97,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 98,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 99,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 100,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 101,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 102,
-	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 103,
-	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 104,
-	PROGRESS_PPBOOT_START = 105,
-	PROGRESS_PPBOOT_FINISH = 106,
-	PROGRESS_BRINGUP_RAM_PRI = 107,
-	PROGRESS_BRINGUP_RAM_PRI_DMA = 108,
-	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 109,
-	PROGRESS_BRINGUP_RAM_SEC = 110,
-	PROGRESS_BRINGUP_RAM_SEC_DMA = 111,
-	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 112,
-	PROGRESS_ARCBOOT_RAM_START = 113,
-	PROGRESS_ARCBOOT_RECOVERY = 114,
-	PROGRESS_ARCBOOT_RECOVERY_DMA = 115,
-	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 116,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 117,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 118,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 119,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 120,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 121,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 122,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 123,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 124,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 125,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INTER_DIE_SYNC = 126,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 127,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 128,
-	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 129,
-	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 130,
-	PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE = 131,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 132,
-	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 133,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 134,
-	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 135,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 136,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 137,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 138,
-	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 139,
-	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_OFFSET = 140,
-	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 141,
-	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 142,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 143,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 144,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 145,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 146,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 147,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 148,
-	PROGRESS_ARCBOOT_EROM_IMG0 = 149,
-	PROGRESS_ARCBOOT_EROM_IMG0_LOADADDR_INVALID = 150,
-	PROGRESS_ARCBOOT_EROM_IMG0_STARTADDR_INVALID = 151,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 152,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 153,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 154,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 155,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 156,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 157,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 158,
-	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 159,
-	PROGRESS_ARCBOOT_EROM_IMG0_START = 160,
-	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 161,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START = 38,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FAIL = 39,
+	PROGRESS_BOOT_INFO_IMG_DESC_DATA_SET_FAIL = 40,
+	PROGRESS_BOOT_INFO_SECURITY_DATA_SET_FAIL = 41,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FINISH = 42,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START = 43,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FAIL = 44,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_INVALID = 45,
+	PROGRESS_BOOT_INFO_IMG_DESC_DATA_GET_FAIL = 46,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH = 47,
+	PROGRESS_BRINGUP_PCI_PRI = 48,
+	PROGRESS_BRINGUP_PCI_PRI_LOADADDR_INVALID = 49,
+	PROGRESS_BRINGUP_PCI_PRI_STARTADDR_INVALID = 50,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 51,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 52,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 53,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 54,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 55,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 56,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 57,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 58,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 59,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 60,
+	PROGRESS_BRINGUP_PCI_SEC = 61,
+	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 62,
+	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 63,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 64,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 65,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 66,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 67,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 68,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 69,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 70,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 71,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 72,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 73,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 74,
+	PROGRESS_ARCBOOT_ROM_TPM = 75,
+	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 76,
+	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 77,
+	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 78,
+	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 79,
+	PROGRESS_BRINGUP_TPM_SPI = 80,
+	PROGRESS_BRINGUP_TPM_SPI_PRI = 81,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_LOADADDR_INVALID = 82,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_STARTADDR_INVALID = 83,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 84,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 85,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 86,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 87,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 88,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 89,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 90,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 91,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 92,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 93,
+	PROGRESS_BRINGUP_TPM_SPI_SEC = 94,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_LOADADDR_INVALID = 95,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_STARTADDR_INVALID = 96,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 97,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 98,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 99,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 100,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 101,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 102,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 103,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 104,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 105,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 106,
+	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 107,
+	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 108,
+	PROGRESS_PPBOOT_START = 109,
+	PROGRESS_PPBOOT_FINISH = 110,
+	PROGRESS_BRINGUP_RAM_PRI = 111,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 112,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 113,
+	PROGRESS_BRINGUP_RAM_SEC = 114,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 115,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 116,
+	PROGRESS_ARCBOOT_RAM_START = 117,
+	PROGRESS_ARCBOOT_RECOVERY = 118,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 119,
+	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 120,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 121,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 122,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 123,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 124,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 125,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 126,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 127,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 128,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 129,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INTER_DIE_SYNC = 130,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 131,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 132,
+	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 133,
+	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 134,
+	PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE = 135,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 136,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 137,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 138,
+	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 139,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 140,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 141,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 142,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 143,
+	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_OFFSET = 144,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 145,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 146,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 147,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 148,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 149,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 150,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 151,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 152,
+	PROGRESS_ARCBOOT_EROM_IMG0 = 153,
+	PROGRESS_ARCBOOT_EROM_IMG0_LOADADDR_INVALID = 154,
+	PROGRESS_ARCBOOT_EROM_IMG0_STARTADDR_INVALID = 155,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 156,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 157,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 158,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 159,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 160,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 161,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 162,
+	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 163,
+	PROGRESS_ARCBOOT_EROM_IMG0_START = 164,
+	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 165,
 };
